packages/manager/apps/pci-private-network/src/pages/new/steps/ConfigurationStep.tsx

```diff
@@ -220,17 +219,31 @@ export default function ConfigurationStep({
   const missingNameError =
     isNetworkNameInputTouched && !store.form.privateNetworkName.length;
 
+  const isCidrIpAddressValid = isValidIpAddress(store.form.address);
+  const isCidrMaskValid = isValidCidrMask(store.form.cidr);
+
+  const privateNetworkIsEmpty = () => !store.form.privateNetworkName;
+  const createGatewayIsCheckAndProjectIsDiscovery = () =>
+    !store.form.createGateway && store.project.isDiscovery;
+  const dhcpIsCheckAndCidrNotValid = () =>
+    store.form.dhcp && (!isCidrIpAddressValid || !isCidrMaskValid);
+
+  const hadToDisabledCreateButton =
+    privateNetworkIsEmpty() ||
+    createGatewayIsCheckAndProjectIsDiscovery() ||
+    dhcpIsCheckAndCidrNotValid();
```

Comment:
**nitpick:** Avoid recomputing. Moreover the variable names used are very complicated to interpret. If there's a need to use them, simplify the variable names.